src/main/java/org/jahia/modules/appshell/Main.java

```diff
@@ -144,49 +218,119 @@ private void setCustomAttributes(JahiaUser currentUser, HttpServletRequestWrappe
         wrapper.setAttribute("config", config.toString());
     }
 
+    public void updateBundleInfos() {
+        List<Bundle> packages = getPackages();
+
+        List<Pattern> newUrlPatterns = new ArrayList<>();
+        Map<Bundle, JSONObject> newBundleDescriptors = new LinkedHashMap<>();
+
+        for (Bundle bundle : packages) {
+            JSONObject pkgJson = null;
+            try {
+                pkgJson = new JSONObject(IOUtils.toString(bundle.getResource(PACKAGE_JSON) != null ? bundle.getResource(PACKAGE_JSON) : bundle.getResource(JAHIA_JSON)));
+                if (pkgJson.has(JAHIA) && pkgJson.getJSONObject(JAHIA).has(URLPATTERNS)) {
+                    Object urlPatternsObj = pkgJson.getJSONObject(JAHIA).get(URLPATTERNS);
+                    List<String> urlPatternArray = getStringList(urlPatternsObj);
+                    for (String urlPattern : urlPatternArray) {
+                        Pattern compiledUrlPattern = Pattern.compile(urlPattern);
+                        newUrlPatterns.add(compiledUrlPattern);
+                    }
+
+                }
+                newBundleDescriptors.put(bundle, pkgJson);
+            } catch (JSONException | IOException e) {
+                logger.warn("Error processing bundle JSON descriptor for bundle {}", bundle, e);
```

Comment:
This look more than a warn in case it happen, and also I'm not sure that Bundle object is well represented when concat with a String.
I think the best is to not catch this errors and let the code crash higher as it could lead to inconsistency in the resulted UI is bundle scripts are not loaded correctly.

Reply:
I don't understand. If we let the exception go higher, then all the descriptors of all the bundles would not be updated, which could also be a problem. Shouldn't we just fail on the single one ?